src/api/c/index.cpp

```diff
@@ -36,6 +36,38 @@ static void indexArray(af_array &dest, const af_array &src, const unsigned ndims
     dest = getHandle(dst);
 }
 
+af_err af_make_array_index(af_index_t** result, const af_array in)
+{
+    try {
+        if(*result) {
+            // avoid a dangling pointer
+            return AF_ERR_NOT_SUPPORTED;
+        }
+        *result = new af_index_t;
+        (*result)->idx.arr = in;
+        (*result)->isBatch = true;
```

Comment:
Oh I get it now. isSeq has sufficient information to tell if it is an array or seq.

Should we have an additional parameter then in the functions? i.e. 
```c++
af_err af_make_array_index(af_index_t** result, const af_array in, bool isBatch)
```